### PR TITLE
Add update documentation warning for non-public functions

### LINKED ISSUE #29579: Update documentation warning/code action not shown for non-public functions
**Description:**
Consider the following function:
```ballerina
# Description
#
# + name - My Name
function myFunc(string name, string val) {
    
}
```
Here' the 2nd parameter `val` is not suggested to be added to the documentation via `update documentation` code action. Or the usual warning of `undocumented parameter` is not shown as well. However the same works for `public` functions.

When discussed with @pubudu91, the documentation diagnostics for non-public functions are not generated at the moment because they are not visible in API docs. For better usability, this needs to be supported for non-public functions as well.

**Steps to reproduce:**
See description

**Affected Versions:**
SLA3

**Suggested Labels (optional):**
SemanticAPI?

### PR DESCRIPTION
**Pull request by suleka96**
Fixes #29579

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DocumentationAnalyzer.java [hunk 1]
```diff
@@ -485,12 +484,10 @@ private void validateParameters(DocumentableNode documentableNode,
                     dlog.warning(((BLangNode) parameter).pos, undocumentedParameter, name);
                 }
 
-                // If the parameter is a public function parameter, the parameter should be documented.
+                // If the parameter is public or non-public function parameter,
+                // the parameter should be documented.
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DocumentationAnalyzer.java:488]: We don't need this comment anymore, do we?

**Review by MaryamZi — COMMENTED** [review]
Shall we also add a test for the compiler, to test the warning?

File: language-server/modules/langserver-core/src/test/resources/codeaction/update-documentation/source/updateDocumentation3.bal [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+# This function has a doc with multiple lines. Once updated, it should
+# preserve these lines
+#
```
Comment (MaryamZi) [inline language-server/modules/langserver-core/src/test/resources/codeaction/update-documentation/source/updateDocumentation3.bal:4]: Are these relevant to this particular test?

Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/test/resources/codeaction/update-documentation/source/updateDocumentation3.bal:4]: These are okay.

Comment (IMS94) [reply language-server/modules/langserver-core/src/test/resources/codeaction/update-documentation/source/updateDocumentation3.bal:4]: @MaryamZi Yes. This test case is to test the ability to retain multiline descriptions when `update documentation` code action is being used.

**Review by IMS94 — DISMISSED** [review]
LGTM! There's a conflict to resolve.

**Review by MaryamZi — APPROVED** [review]
